index.js

```diff
@@ -37,9 +38,25 @@ const fastifyAutoload = async function autoload (fastify, options) {
       })
   }))
 
-  for (const name in pluginsMeta) {
-    const plugin = pluginsMeta[name]
-    registerPlugin(fastify, plugin, pluginsMeta)
+  if (!options.autoHooks) { // default behaviour: apply plugins at root level of fastify app, no encapsulation required
```

Comment:
I think we might remove this branch altogether.

Reply:
I agree, that will make things a lot easier to follow.

I will refactor this section and add a few benchmarks to make sure this isn't adding any significant overhead.